### DISCUSSION TITLE
TypeError: Cannot read private member #headersList from an object whose class did not declare it

### DISCUSSION THREAD

**Discussion by kumargauravin** [vercel/next.js#68849] [App Router]
Using App Router when using next/headers getting this on Node 20.16.0 Linux EC2 instance. 

  const headersList = headers();

this is what I am using in page.tsx

getting this on server. Server is expecting headers from SSO FR. Next 14.2.3 React 18.3.1

0|dashboar | TypeError: Cannot read private member #headersList from an object whose class did not declare it
0|dashboar |     at [nodejs.util.inspect.custom] (node:internal/deps/undici/undici:8647:64)
0|dashboar |     at formatValue (node:internal/util/inspect:806:19)
0|dashboar |     at inspect (node:internal/util/inspect:365:10)
0|dashboar |     at formatWithOptionsInternal (node:internal/util/inspect:2304:40)
0|dashboar |     at formatWithOptions (node:internal/util/inspect:2166:10)
0|dashboar |     at console.value (node:internal/console/constructor:349:14)
0|dashboar |     at console.log (node:internal/console/constructor:386:61)
0|dashboar |     at d (/apps/opt/data/sor-nbs-recon-dashboard-service-ui/dist/server/app/new-user/page.js:1:2693)
0|dashboar |     at eh (/apps/opt/data/sor-nbs-recon-dashboard-service-ui/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:134786)
0|dashboar |     at e (/apps/opt/data/sor-nbs-recon-dashboard-service-ui/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:137671) {
0|dashboar |   digest: '463973562'
0|dashboar | }

**Comment by Juneezee** [comment]
Do you also get the error locally with `next dev`?

**Reply by kumargauravin** [reply]
That is what, does not comes in local though I found solution. If I do console.log of object like headers list in this case it fails due to Typescript rule

**Reply by StephenTangCook** [reply]
I started seeing this error today as well. @kumargauravin are you saying that `console.log` of the headers actually fixes the issue?? 😖

**Reply by kumargauravin** [reply]
No I said do not use the object directly use its functions to iterate error comes when for example you are trying to console.log(headersList) in this case

**Reply by kumargauravin** [reply]
If you use headersList.(iterative functions) provided by Headers it is all working good, something to do with how a static member of a class has to be exposed seems like more abstraction added.

**Reply by StephenTangCook** [reply]
Ah, I see! Thanks!

**Reply by mobeigi** [reply]
Ran into same issue, all I did was try to `console.log()` the headers in a server component.

As a workaround this works without throwing the exception:
```ts
const headerList = headers();
for (const [key, value] of headerList.entries()) {
  console.log(`${key}: ${value}`);
}
```

**Comment by ARiyou2000** [comment]
I still have the problem even after doing all that. What is the actual fix?

**Reply by kumargauravin** [reply]
Can you share your snippet! Whatever property you want to read from Headers do not use directly but using functions exposed by the instance.

**Comment by nyeoni** [comment]
I resolved the issue by removing console.log('[Page] headers', headers()). Thanks!

**Comment by kumargauravin** [comment]
Hi,

Yes as headers have private members which were getting accessed. If you
need you can iterate using functions provided than console will also work.

Regards,
Kumar

**Comment by rodrigopa** [comment]
```Object.fromEntries(await headers());```

**Reply by rodrigopsofto** [reply]
or
```const headersList = new Headers(await headers());```

**Comment by MichaelTendoSsemwanga** [comment]
+1
